src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -386,6 +376,26 @@ private function addServiceToAmbiguousType($id, $type)
         $this->ambiguousServiceTypes[$type][] = $id;
     }
 
+    /**
+     * @param \ReflectionClass $reflectionClass
+     *
+     * @return \ReflectionMethod[]
+     */
+    private static function getSetters(\ReflectionClass $reflectionClass)
+    {
+        foreach ($reflectionClass->getMethods(\ReflectionMethod::IS_PUBLIC) as $reflectionMethod) {
+            if (
```

Comment:
btw, not sure continue+yield is better, I'd move yield into the `if` instead personally